src/ui/features/onboarding/Backup/Backup.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { accountPublicRPCPort, walletPort } from 'src/ui/shared/channels';
+import { setCurrentAddress } from 'src/ui/shared/requests/setCurrentAddress';
+import { invariant } from 'src/shared/invariant';
+import { getWalletGroupByAddress } from 'src/ui/shared/requests/getWalletGroupByAddress';
+import { SessionExpired } from 'src/shared/errors/errors';
+import { BackupComponent } from 'src/ui/pages/Backup/Backup';
+import { useNavigate } from 'react-router-dom';
+import { useMutation } from '@tanstack/react-query';
+import { isSessionExpiredError } from 'src/ui/shared/isSessionExpiredError';
+
+async function ensurePendingWalletAndUser() {
+  const wallet = await walletPort.request('getPendingWallet');
+  if (!wallet) {
+    throw new SessionExpired();
+  }
+  await accountPublicRPCPort.request('saveUserAndWallet');
+  await setCurrentAddress({ address: wallet.address });
+  return wallet;
+}
+
+export function Backup() {
+  const navigate = useNavigate();
+
+  const handleError = (error: unknown) => {
+    if (isSessionExpiredError(error)) {
+      navigate('/onboarding/session-expired', { replace: true });
+    }
+  };
+
+  const { mutate: handleSkip } = useMutation({
+    mutationFn: () => ensurePendingWalletAndUser(),
+    onSuccess: () => navigate('/onboarding/success'),
+    onError: handleError,
+    useErrorBoundary: true,
+  });
+
+  const { mutate: handleSuccess } = useMutation({
+    mutationFn: async () => {
+      const wallet = await ensurePendingWalletAndUser();
+      const group = await getWalletGroupByAddress(wallet.address);
+      invariant(group?.id, 'Saved wallet should belong to a group');
+      await walletPort.request('updateLastBackedUp', { groupId: group.id });
+    },
+    onSuccess: () => navigate('/onboarding/success'),
+    onError: handleError,
+    useErrorBoundary: true,
+  });
+
+  return (
+    <BackupComponent
+      onStart={() => navigate('/onboarding/backup/recovery-phrase')}
+      onExit={() => navigate('/onboarding')}
+      onSkip={handleSkip}
+      onSuccess={handleSuccess}
+      onSessionExpired={() =>
+        navigate('/onboarding/session-expired', { replace: true })
+      }
```

Comment:
Since this value is used as a dependency of `useEffect`, we should pass a stable value, e.g. wrap in `useCallback`:

https://github.com/zeriontech/zerion-wallet-extension/pull/569/files#diff-04ff63434a4618dc184dc118c89196327a3a3f3575863a2fc7a2ab11755a00ffR47

Reply:
Thanks, yeah, both `RecoveryPhrase` and `VerifyBackup` has it as a dependency of `useEffect`, updated ✅